pkg/thanos/operator.go

```diff
@@ -703,20 +703,30 @@ func (o *Operator) UpdateStatus(ctx context.Context, key string) error {
 	return nil
 }
 
-func createSSetInputHash(tr monitoringv1.ThanosRuler, c Config, ruleConfigMapNames []string, ss interface{}) (string, error) {
+func createSSetInputHash(tr monitoringv1.ThanosRuler, c Config, ruleConfigMapNames []string, ss appsv1.StatefulSetSpec) (string, error) {
 	hash, err := hashstructure.Hash(struct {
-		TR monitoringv1.ThanosRuler
-		C  Config
-		S  interface{}
-		R  []string `hash:"set"`
-	}{tr, c, ss, ruleConfigMapNames},
+		ThanosRulerLabels      map[string]string
+		ThanosRulerAnnotations map[string]string
+		ThanosRulerGeneration  int64
+		ThanosRulerSpec        monitoringv1.ThanosRulerSpec
```

Comment:
you can remove the `ThanosRulerSpec ` field (a change in the generation value implies an update of the spec).

Reply:
Prometheus and AlertManager both use Generation as well as spec

Reply:
unless I'm mistaken, they use the statefulset's spec not the Prometheus/Alertmanager spec?

Reply:
You're right, I need to have my eyes checked.